docs/pairing-architecture.md

```diff
@@ -0,0 +1,282 @@
+## Pairing Flow Architecture
+
+The Desktop and Mobile devices will not communicate directly, because we cannot assume that they will be on the same network and because this is difficult to achieve from web content. Instead, they will exchange encrypted messages via websocket to a lightweight "channel server" service which can relay messages back and forth:
+
+
+![](images/pairing-flow1.png)
+
+*****
+
+
+![](images/pairing-flow2.png)
+
+### Detailed Flow
+
+Is is advised to read the following section while looking at the detail digram while reading the information below.
+
+![](https://www.lucidchart.com/publicSegments/view/00b7d8a7-fd78-493f-91f5-120b4034f8b0/image.png)
+
+[View / Edit in LucidChart](https://www.lucidchart.com/documents/edit/1bc1b604-0047-4542-8827-ed8518b0433e/0)
+
+
+#### 1) Desktop opens an encrypted channel and advertises it via QR code
+
+To make itself available for pairing, the Desktop device opens a connection to the Channel Server using the [fxa-pairing-channel library](https://github.com/mozilla/fxa-pairing-channel). 
+Any application that learns the channel key will be able to request pairing from the Desktop device, or intercept messages exchanges during the pairing flow.
+This library uses a subset of TLS-like communication between the two devices.
```

Comment:
`a subset of TLS-like` :)

Reply:
One of the outcomes of the recent secreview was that saying "subset" and "TLS-like" causes concern because it's easy to interpret them as meaning "we removed some stuff from TLS because we didn't think it was necessary" and then they wonder what things were removed and whether it was actually important for security...

Can we say something like "the pre-shared key more of TLS1.3" to be more precise in scope?

Reply:
typo `more` --> `mode` ?